physics/cires_ugwp_module_v1.F90

```diff
@@ -0,0 +1,666 @@
+   real, parameter :: cd_ulim = 1.0                 ! critical level precision or Lz ~ 0 ~dz of model
+
+   contains
+!
+! -----------------------------------------------------------------------
+!
+! init  of cires_ugwp   (_init)  called from CCPP cap file
+!
+! -----------------------------------------------------------------------
+
+
+
+  subroutine cires_ugwp_init_v1 (me, master, nlunit, logunit, jdat_gfs, fn_nml2, &
+              lonr, latr, levs, ak, bk, pref, dtp, cdmvgwd, cgwf,                &
+              pa_rf_in, tau_rf_in)
+!
+!  input_nml_file ='input.nml'=fn_nml   ..... OLD_namelist and cdmvgwd(4) Corrected Bug Oct 4
+!
+    use  netcdf
+    use  ugwp_oro_init_v1,     only :  init_oro_gws
+    use  ugwp_conv_init_v1,    only :  init_conv_gws
+    use  ugwp_fjet_init_v1,    only :  init_fjet_gws
+    use  ugwp_okw_init_v1,     only :  init_okw_gws
+    use  ugwp_wmsdis_init_v1,  only :  initsolv_wmsdis
+    
+    use  ugwp_lsatdis_init_v1, only :  initsolv_lsatdis
+       
+    
+    use  ugwp_wmsdis_init_v1,  only : ilaunch, nslope, lhmet, lzmax, lzmin, lzstar   
+    use  ugwp_wmsdis_init_v1,  only : tau_min, tamp_mpa    
+    implicit none
+
+    integer, intent (in) :: me
+    integer, intent (in) :: master
+    integer, intent (in) :: nlunit
+    integer, intent (in) :: logunit
+    integer, intent (in) :: lonr
+    integer, intent (in) :: levs
+    integer, intent (in) :: latr
+    integer, intent (in) :: jdat_gfs(8)
+    real,    intent (in) :: ak(levs+1), bk(levs+1), pref
+    real,    intent (in) :: dtp
+    real,    intent (in) :: cdmvgwd(2), cgwf(2)             ! "scaling" controls for "old" GFS-GW  dims(2) !!!
+    real,    intent (in) :: pa_rf_in, tau_rf_in
+ 
+    character(len=64), intent (in) :: fn_nml2
+    character(len=64), parameter   :: fn_nml='input.nml'
+
+!    character,  intent (in) :: input_nml_file
+!    integer, parameter :: logunit =  6
+    integer :: ios
+    logical :: exists
+    real    :: dxsg
+    
+    integer :: ncid,  iernc, vid, dimid, status         
+    integer :: k
+    integer :: ddd_ugwp,    curday_ugwp 
+    real    :: avqbo(6)
+    avqbo = [0.05, 0.1, 0.25, 0.5, 0.75, 0.95]
+!
+    if (me == master) print *, trim (fn_nml), ' GW-namelist file '
+    inquire (file =trim (fn_nml) , exist = exists)
+!
+    if (.not. exists) then
+       if (me == master) &
+        write (6, *) 'separate ugwp :: namelist file: ', trim (fn_nml), ' does not exist'
+    else
+        open (unit = nlunit, file = trim(fn_nml), action = 'read', status = 'old', iostat = ios)
+    endif
+    rewind (nlunit)
+    read   (nlunit, nml = cires_ugwp_nml)
+    close  (nlunit)
+!
+    strsolver= knob_ugwp_orosolv     
+    pa_rf   = pa_rf_in
+    tau_rf  = tau_rf_in
+    
+    curday_ugwp = jdat_gfs(1)*10000 + jdat_gfs(2)*100 +jdat_gfs(3)    
+    call calendar_ugwp(jdat_gfs(1), jdat_gfs(2), jdat_gfs(3), ddd_ugwp)
+        
+! write version number and namelist to log file
```

Comment:
Quite a lot of output from this routine. Can we reduce it after the initial development and testing phase?

Reply:
Yes we can.